reverse_words/README.rst

```diff
@@ -0,0 +1,119 @@
+Learn to reverse the letters of each individual word in a text
+================================================================
+
+In this demo, a recurrent network equipped with an attention mechanism
+learns to reverse each word (on a character-by-character basis) in its input text. 
+
+The default training data is the Google Billion Word corpus, 
+which you should download and put to the path indicated in your .fuelrc file.
+
+* http://blocks.readthedocs.org/en/latest/
+* http://fuel.readthedocs.org/en/latest/ is practically empty, better to look 
+  at the source : https://github.com/mila-udem/fuel
+
+As an example, the first sentence should be transformed into the second:
+
+* ``The quick brown fox jumps over the lazy dog.`` 
+* ... model ...
+* ``eht kciuq nworb xof spmuj revo eht yzal god.`` 
+
+
+The bulk of the functionality of the code is in the ``__init__.py`` file.
+
+Overall, the model described seems to be as in the paper by Dzmitry Bahdanau, Kyunghyun Cho, Yoshua Bengio (2014) :
+
+.. figure:: 1409.0473v6.Figure1.438x549.png
+   :width: 219px
+   :align: center
+   :alt: Figure 1
+
+   Figure 1 of `Neural Machine Translation by Jointly Learning to Align and Translate <http://arxiv.org/abs/1409.0473>`_
+
+
+Structure of the Data
+--------------------------
+
+The input data arrives through the a ``fuel`` data processing pipeline,
+and the first section of the code deals with functions that clean and prepare the
+data :
+
+* converting character codes to and from a numerical (integer) encoding 
+* a 'gold standard' reverse_words function that performs the task perfectly
+* ``_lower`` and ``_filter_long`` data-cleaning functions that ensure that the 
+  pipeline produces one lower-case character array for each input sentence,
+  and that sentences of over 100 characters are filtered out
+* ``_transpose`` is a convenience function to re-orient the matrices finally coming out of the data stream
+* ``_is_nan`` is a termination function for training - not data processing
```

Comment:
Here I would also mention that putting such functions in the global namespace was done in order to ensure that the final main loop can be pickled.